instance descriptions for cloud instances

trevj opened the pull request:
Depends on:
https://github.com/uProxy/uproxy-docker/pull/35

Name is still whatever hostname or IP is set in the invite code but I think being able to configure the description is a step forward.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy-lib/324)

<!-- Reviewable:end -->

trevj committed:
reject when zork cannot be contacted, rather than crashing the module

trevj committed:
fetch /banner on each reconnect and use it for description

trevj committed:
Merge branch 'dev' into trevj-banner

trevj committed:
change type of saved contacts, to allow on-demand attributes such as server description be cached between logins

trevj committed:
make getBanner reject when there is any error reading the banner

lucyhe commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[src/cloud/social/provider.ts, line 230 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/324#-K3aI5wGiEd7DU5VOlbY:-K3aI5wHzRg2wwLotJwB:-1447752287) ([raw file](https://github.com/uproxy/uproxy-lib/blob/60994275de497718e3ff45722b51ae693c141aa8/src/cloud/social/provider.ts#L230)):_
Does this mean check if there's a banner, and use that for the description?

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/324)_
<!-- Sent from Reviewable.io -->

trevj committed:
remove done TODO

trevj commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[src/cloud/social/provider.ts, line 230 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy-lib/324#-K3aI5wGiEd7DU5VOlbY:-K3aKipz1oyNdUxds7ij:348312724) ([raw file](https://github.com/uproxy/uproxy-lib/blob/60994275de497718e3ff45722b51ae693c141aa8/src/cloud/social/provider.ts#L230)):_
Good catch! That was from a previous iteration of the pull request...which I implemented, but forgot to remove the TODO.

Removed!

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/324)_
<!-- Sent from Reviewable.io -->

lucyhe commented:
hurray :+1: !
